### ISSUE TITLE
[Bug] [Android] [Shell] Toolbar items: text and icon bindings don't update visually

### ISSUE THREAD

**Issue by JeroenTrivec** [xamarin/Xamarin.Forms#8553]
### Description
Tested on an Android Phone running Android 6.0 and an Android emulator running Android 9.0.
This issue does not seem to occur on iOS.

When binding the `Text` or `IconImageSource` properties of a `ToolbarItem` and triggering a change of those properties, the new text/icon is not shown. 

### Steps to Reproduce

1. Create a Shell app. Add a ToolbarItem to your main page with a binding on either its Text or IconImageSource property.
2. Add a Button which changes the text or icon of the ToolbarItem when pressed, and raises the PropertyChanged event on it.
3. Run the app and press the button.

### Expected Behavior
The ToolbarItem shows the new text/icon.

### Actual Behavior
The ToolbarItem's appearance does not change.

### Basic Information

- Version with issue: 4.3.0.991211 and 4.4.0.991210-pre2
- Last known good version: none
- IDE: Visual Studio 2019 (16.3.7)
- Platform Target Frameworks: 
  - Android: 9.0 

### Screenshots

In the following screenshot, the left ToolbarItem's Text is bound to the same property as the Label beneath the Button, and the second ToolbarItem's IconImageSource has the same binding as the Image below the Label. The increment button increments the number and toggles between two icons. It was pressed once.

![device-2019-11-18-160928](https://user-images.githubusercontent.com/40200871/69064103-efb27200-0a1d-11ea-8a7a-f8ddf95ca2da.png)

### Reproduction Link

[ToolbarItemBug.zip](https://github.com/xamarin/Xamarin.Forms/files/3859275/ToolbarItemBug.zip)

### Extra info

We ran into this issue after converting our app from a MasterDetailPage-based app to the Shell. The old version of the app never had this issue.

**Comment by smitter86** [comment]
I can confirm that I have similar issue running on XF 4.3.0.947036

**Comment by jfversluis** [comment]
I confirmed that this is not working with any of the packages available right now. However, switching to our nightly feed, I find that the latest package there actually does have this working.

Searching a bit more on GitHub I found that this is a duplicate of #7181 and is fixed in 4.3 and up. A new 4.3 and 4.4 release should be available any day now so be on the lookout for that one :)

Closing this as a duplicate for now. Thanks!
